packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -226,6 +227,11 @@ export class Trail extends Resource {
       },
     }));
 
+    this.topic = props.snsTopic;
+    if (this.topic) {
+      this.topic.grantPublish(cloudTrailPrincipal);
+    }
```

Comment:
if you agree with not making it a property of the class, this could become
```suggestion
    props.topic?.grantPublish(cloudTrailPrincipal)
```